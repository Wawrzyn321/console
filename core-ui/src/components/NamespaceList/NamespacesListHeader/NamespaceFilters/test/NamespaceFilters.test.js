import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import NamespaceFilters from './../NamespaceFilters';

describe('NamespaceFilters', () => {
  it('Renders search labels', () => {
    //for Popovers's Warning: `NaN` is an invalid value for the `left` css style property.
    console.error = jest.fn();

    const { queryByText, queryByLabelText } = render(
      <NamespaceFilters
        filters={[{ name: 'first-label' }, { name: 'second-label' }]}
        updateFilters={() => {}}
      />,
    );

    const labelsButton = queryByLabelText('open-filters');
    expect(labelsButton).toBeInTheDocument();
    fireEvent.click(labelsButton);

    ['first-label', 'second-label'].forEach(name =>
      expect(queryByText(name)).toBeInTheDocument(),
    );
  });
});
