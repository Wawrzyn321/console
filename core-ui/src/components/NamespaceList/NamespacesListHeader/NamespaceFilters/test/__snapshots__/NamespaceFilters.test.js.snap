// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamespaceFilters Renders with minimal props 1`] = `
<div
  className="fd-popover"
>
  <div
    onFocus={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="fd-popover__control"
    >
      <button
        aria-controls="fd-jxB_s4IX"
        aria-expanded={false}
        aria-haspopup={true}
        className="fd-button--light sap-icon--filter"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
  </div>
</div>
`;
